### ISSUE TITLE
Finetuning Deep Learning IQA Models with a Custom Dataset

### ISSUE THREAD

**Issue by Stevieee83** [chaofengc/IQA-PyTorch#87]
Hello, I have a question regarding the deep learning IQA models in the library. Is it possible to fine-tune a model on a custom dataset using the respective .yml config file for the any of the models? I am able to train the models on the LIVEC dataset as in the examples but when I use a custom dataset for any of the trainable models I am getting errors in the terminal from multiple .py files in the IQA-PyTorch library.

Thank you for your time and I look forward to your response.

**Comment by chaofengc** [comment]
Sorry that there is no easy way to fine-tune a model without enough knowledge about the dataset and method itself. 

You have to rewrite the dataset loader with examples from `pyiqa/data` and model training pipeline with examples from `pyiqa/models` for your own tasks.

**Comment by Stevieee83** [comment]
Thank you for your response. I used the examples to rewrite the dataset but the Python interpreter is giving me the following error:

raise KeyError(f"No object named '{name}' found in '{self._name}' registry!")
KeyError: "No object named 'MyDataset' found in 'dataset' registry!"

I have added the MyDataset object to the pyiqa/data directory as a .py Python file but the interpreter is telling me the object is not in the dataset registry. I used the livechallenge_dataset.py example file to create an object for the dataset by changing the class name to MyDataset. MyDataset has been added to the class constructor too. The .yml config file is configured to fine-tune the DBCNN model but I am getting the error from the Python interpreter every time.

Thank you for your time and I look forward to your response.

**Comment by chaofengc** [comment]
You might miss two points:
- The dataset `.py` file should be named as `xxx_dataset.py`
- The dataset class should be registered with decorator, such as:
   https://github.com/chaofengc/IQA-PyTorch/blob/9dc09cfa496da1c34d7f5ead933749b3b49716e3/pyiqa/data/livechallenge_dataset.py#L17

**Comment by Stevieee83** [comment]
Thank you for your response. The .py file is named mydataset_dataset.py, and the dataset class includes the @DATASET_REGISTRY.register() decorator code above the MyDataset class object in line 17 of mydataset_dataset.py. I copied the code from the original livechallenge_dataset.py Python file. Is there something else I have to do in the library to get my new dataset Python class to work?

I checked everthing again and I am still getting the same error from the Python interpretor:

raise KeyError(f"No object named '{name}' found in '{self._name}' registry!")
KeyError: "No object named 'MyDataset' found in 'dataset' registry!"

I look forward to your response, and thanks again for your time.
